src/main/scala/is/hail/methods/SampleQC.scala

```diff
@@ -163,52 +152,68 @@ class SampleQCCombiner(val keepStar: Boolean) extends Serializable {
       divNull(nTi, nTv),
       divNull(nHet, nHomVar),
       divNull(nIns, nDel))
+  }
 }
 
 object SampleQC {
-  def results(vds: GenericDataset, keepStar: Boolean): Map[Annotation, SampleQCCombiner] = {
-    val extract = Genotype.buildGenotypeExtractor(vds.genotypeSignature)
-
+  def results(vds: GenericDataset): Map[Annotation, SampleQCCombiner] = {
```

Comment:
Can we just return an `Array[SampleQCCombiner]` and add a `VSM.anntotateSamples(path, signature, newAnnotation: Array[Annotation])`?  This map is ancient silliness.